native/dlarfx.go

```diff
@@ -0,0 +1,527 @@
+			}
+			return
+
+		case 8: // Special code for 8×8 Householder matrix.
+			v0 := v[0]
+			t0 := tau * v0
+			v1 := v[1]
+			t1 := tau * v1
+			v2 := v[2]
+			t2 := tau * v2
+			v3 := v[3]
+			t3 := tau * v3
+			v4 := v[4]
+			t4 := tau * v4
+			v5 := v[5]
+			t5 := tau * v5
+			v6 := v[6]
+			t6 := tau * v6
+			v7 := v[7]
+			t7 := tau * v7
+			for j := 0; j < n; j++ {
+				sum := v0*c[j] + v1*c[ldc+j] + v2*c[2*ldc+j] + v3*c[3*ldc+j] + v4*c[4*ldc+j] +
+					v5*c[5*ldc+j] + v6*c[6*ldc+j] + v7*c[7*ldc+j]
+				c[j] -= sum * t0
+				c[ldc+j] -= sum * t1
+				c[2*ldc+j] -= sum * t2
+				c[3*ldc+j] -= sum * t3
+				c[4*ldc+j] -= sum * t4
+				c[5*ldc+j] -= sum * t5
+				c[6*ldc+j] -= sum * t6
+				c[7*ldc+j] -= sum * t7
+			}
+			return
+
+		case 9: // Special code for 9×9 Householder matrix.
+			v0 := v[0]
+			t0 := tau * v0
+			v1 := v[1]
+			t1 := tau * v1
+			v2 := v[2]
+			t2 := tau * v2
+			v3 := v[3]
+			t3 := tau * v3
+			v4 := v[4]
+			t4 := tau * v4
+			v5 := v[5]
+			t5 := tau * v5
+			v6 := v[6]
+			t6 := tau * v6
+			v7 := v[7]
+			t7 := tau * v7
+			v8 := v[8]
+			t8 := tau * v8
+			for j := 0; j < n; j++ {
+				sum := v0*c[j] + v1*c[ldc+j] + v2*c[2*ldc+j] + v3*c[3*ldc+j] + v4*c[4*ldc+j] +
+					v5*c[5*ldc+j] + v6*c[6*ldc+j] + v7*c[7*ldc+j] + v8*c[8*ldc+j]
+				c[j] -= sum * t0
+				c[ldc+j] -= sum * t1
+				c[2*ldc+j] -= sum * t2
+				c[3*ldc+j] -= sum * t3
+				c[4*ldc+j] -= sum * t4
+				c[5*ldc+j] -= sum * t5
+				c[6*ldc+j] -= sum * t6
+				c[7*ldc+j] -= sum * t7
+				c[8*ldc+j] -= sum * t8
+			}
+			return
+
+		case 10: // Special code for 10×10 Householder matrix.
+			v0 := v[0]
+			t0 := tau * v0
+			v1 := v[1]
+			t1 := tau * v1
+			v2 := v[2]
+			t2 := tau * v2
+			v3 := v[3]
+			t3 := tau * v3
+			v4 := v[4]
+			t4 := tau * v4
+			v5 := v[5]
+			t5 := tau * v5
+			v6 := v[6]
+			t6 := tau * v6
+			v7 := v[7]
+			t7 := tau * v7
+			v8 := v[8]
+			t8 := tau * v8
+			v9 := v[9]
+			t9 := tau * v9
+			for j := 0; j < n; j++ {
+				sum := v0*c[j] + v1*c[ldc+j] + v2*c[2*ldc+j] + v3*c[3*ldc+j] + v4*c[4*ldc+j] +
+					v5*c[5*ldc+j] + v6*c[6*ldc+j] + v7*c[7*ldc+j] + v8*c[8*ldc+j] + v9*c[9*ldc+j]
+				c[j] -= sum * t0
+				c[ldc+j] -= sum * t1
+				c[2*ldc+j] -= sum * t2
+				c[3*ldc+j] -= sum * t3
+				c[4*ldc+j] -= sum * t4
+				c[5*ldc+j] -= sum * t5
+				c[6*ldc+j] -= sum * t6
+				c[7*ldc+j] -= sum * t7
+				c[8*ldc+j] -= sum * t8
+				c[9*ldc+j] -= sum * t9
+			}
+			return
+		}
+	}
+
+	// Form C * H, where H has order n.
+	switch n {
+	default: // Code for general n.
+		impl.Dlarf(side, m, n, v, 1, tau, c, ldc, work)
+		return
+
+	case 0: // No-op for zero size matrix.
+		return
+
+	case 1: // Special code for 1×1 Householder matrix.
+		t0 := 1 - tau*v[0]*v[0]
+		for j := 0; j < m; j++ {
+			c[j*ldc] = t0 * c[j*ldc]
+		}
+		return
+
+	case 2: // Special code for 2×2 Householder matrix.
+		v0 := v[0]
+		t0 := tau * v0
+		v1 := v[1]
+		t1 := tau * v1
+		for j := 0; j < m; j++ {
+			sum := v0*c[j*ldc] + v1*c[j*ldc+1]
```

Comment:
For all of these, consider `cs := c[j*ldc:]`. This code is then more easily vectorized. TODO is fine.